### PR TITLE
Add license type for managed integration runtime

### PR DESCRIPTION
**Pull request by HaoQian-MS**
## Description ##

Add parameter "LicenseType" for "Set-AzureRmDataFactoryV2IntegrationRuntime" cmd.
**Private .Net SDK is included in this PR.**

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [ ] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.
- [ ] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [ ] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [ ] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [ ] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [ ] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [ ] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by markcowl** [comment]
Cmdlet Review is here: https://github.com/Azure/azure-powershell-pr/wiki/ADF-Azure-Hybrid-Use-Benefit#review-notes

**Comment by markcowl** [comment]
@HaoQian-MS Per your cmdlet review, this feature is only available to whitelisted subscriptions, correct?  This means that you should likely release this privately, or as a preview, and keep thsi separate from the stable ADF module that ships with AzureRM.

**Comment by HaoQian-MS** [comment]
Answers to cormacpayne's question:
(1) .Net SDK changes will be made public and pushed to Nuget by 1/18 for Jan release wave. Confirmed with Hermine. I need my change in SDK which add a property to managed integration runtime to avoid build break in powershell side.
(2) (3) (4) Done.

**Comment by HaoQian-MS** [comment]
Reply to markcowl:
We made the decision to release it publicly in Jan. The whitelist(EC) at backend is to make sure customers can enjoy the functionality from powershell and UI at the same time. It contains only test subscriptions before UI is ready. And after that, we won't limit the functionality to certain customers.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 [hunk 1]
```diff
@@ -112,6 +112,7 @@ function Test-SsisAzure-IntegrationRuntime
             -CatalogAdminCredential $mycreds `
             -CatalogPricingTier 'S1' `
             -MaxParallelExecutionsPerNode 1 `
+            -LicenseType LicenseIncluded `
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: @HaoQian-MS it doesn't look like any of the recordings were updated for this test, so the recorded requests and responses won't include this parameter

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md [hunk 2]
```diff
@@ -181,13 +181,27 @@ Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -LicenseType
+The license type that you want to select for the SSIS IR. There are two types: LicenseIncluded or BasePrice. If you are qualified for the Azure Hybrid Use Benefit (AHUB) pricing, please select BasePrice. If not, please select LicenseIncluded.```yaml
```
Comment (cormacpayne) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md]: @HaoQian-MS the trailing ` ```yaml ` block should be moved to a line by itself - this is a bug when using `platyPS`. Please see other parameters in this markdown file for examples.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 (hunk 1 above)
Comment (HaoQian-MS) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: Checked again. This ps function 'Test-SsisAzure-IntegrationRuntime' is not included in test (IntegrationRuntimeTests.cs) yet. So, there's no original recordings for me to update.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md (hunk 2 above)
Comment (HaoQian-MS) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2/help/Set-AzureRmDataFactoryV2IntegrationRuntime.md]: Fixed.

**Comment by cormacpayne** [comment]
@HaoQian-MS 

(1) Please remove the locally built SDK that was added in this PR and update the reference to `Microsoft.Azure.Management.DataFactory` to 0.4.0-preview in your `.csproj` and `packages.config` files

(2) Please open an issue for enabling the disabled test for the next milestone and add the `Test Debt` label to the issue

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/375/

**Pull request merged**